### ISSUE TITLE
Logic in osEventFlagsSet() for xEventGroupSetBitsFromISR() return appears to be inverted

### ISSUE THREAD

**Issue by josephnobes-stfc** [STMicroelectronics/STM32CubeF2#6]
**Set-up**
STM32F205 project on custom board using STM32Cube FW_F2 1.9.2 with FreeRTOS middleware enabled (using CMSIS RTOS v2).

**Issue**
When program comes to execute `osEventFlagsSet()` from within an ISR, it returns `osFlagsErrorResource` after the flag has been created through the IOC editor.

Under investigation, it seems that the logic in the [aforementioned function](https://github.com/STMicroelectronics/STM32CubeF2/blob/master/Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c) for handling the return value of `xEventGroupSetBitsFromISR()` may be inverted. When `pdPass` is returned to [indicate a successful addition of the message to the queue internally](https://www.freertos.org/xEventGroupSetBitsFromISR.html), the resonse of the calling function is to report an error, which should be the response to `pdFail`.

As a comparison, `osEventFlagsClear()` calls `xEventGroupClearBitsFromISR()`, but returns the error if the return value *is* `pdFail`.

Checking the [example code](https://www.freertos.org/xEventGroupSetBitsFromISR.html) published by FreeRTOS, `!= pdFail` test is used to trigger the yield, therefore I think the behaviour should be as follows:

```C
if (xEventGroupSetBitsFromISR (hEventGroup, (EventBits_t)flags, &yield) != pdFAIL) {
    rflags = flags;
    portYIELD_FROM_ISR (yield);
} else {
    rflags = (uint32_t)osErrorResource;
}
```

Thanks for the help. I've check that this is still the case in the latest commit to master and can't find anything else about it. If this is the wrong forum for this suggestion please let me know.

**Comment by ALABSTM** [comment]
Hi @josephnobes-stfc,

Thank you for this clear and detailed report. We will forward it to our development teams and get back to you once we have their feedback.

With regards,

**Comment by HBOSTM** [comment]
Hello @josephnobes-stfc,

Thank you for this contribution, this point has been reported to our development teams. I will get back to you as soon as I have any updates.

Best Regards,

**Comment by HBOSTM** [comment]
ST Internal Reference:  152007

**Comment by TOUNSTM** [comment]
Hello @josephnobes-stfc,

I hope you are fine. The point you have raised has been fixed in the [CMSIS-FreeRTOS v10.2.0](https://github.com/ARM-software/CMSIS-FreeRTOS/commit/7823ec00c99af0b5aaa49b9d85eb199b475e3d75). Unfortunately, there is no plan to upgrade the FreeRTOS version for STM32CubeF2 . 

Please allow me to close this issue. Thank you for your comprehension and thank you again for having reported the point.

With Regards,
